Persephone/Controllers/AlbumViewController.swift

```diff
@@ -0,0 +1,81 @@
+//
+//  AlbumViewController.swift
+//  Persephone
+//
+//  Created by Daniel Barber on 2019/2/08.
+//  Copyright © 2019 Dan Barber. All rights reserved.
+//
+
+import Cocoa
+
+class AlbumViewController: NSViewController,
+                           NSCollectionViewDataSource,
+                           NSCollectionViewDelegate,
```

Comment:
From a SOLID perspective, I prefer that the view controller & data source are separate objects. That way you have an object whose single responsibility is data management, and another whose responsibility is view management. To start I usually keep the view controller as its' own delegate, and separate _that_ into a separate object if the delegate functionality becomes large or complicated.

What do you think about that approach?

Reply:
I agree with this, but I struggled to figure what to call those objects and where to put them. Any advice on this would be appreciated.

Reply:
Perhaps I would have a `DataSources` group and a `Delegates` group?

Reply:
Yeah that seems reasonable to me!

Reply:
Perhaps if you have some time on Friday you could go through this with me? I'm not sure exactly how to split this stuff out.